### PR TITLE
[Core/Slider] Slider now supports vertical orientation 

### LINKED ISSUE #1645: [Slider] Add support for vertical slider orientation
Hi Blueprint Team,

It would be awesome if you could come out with a vertical slider along with the horizontal one. Use cases involve volume level, high-to-low index, etc...

Thanks

### PR DESCRIPTION
**Pull request by cmslewis**
#### Checklist

- [x] Include tests

#### Changes proposed in this pull request:

- 🎉 __NEW__ `Slider` now supports vertical orientation!
    - Reworked existing SCSS mixins to contain only orientation-specific code (before, they were decomposed despite being used in only one place.)
    - Reworked SCSS variables but kept aliases to the old ones (`$handle-height`, `$track-height`).
    - Min/default height is `150px`, the same as the current `min-width` for sliders.

#### Reviewers should focus on:

- We don't export mixins, right? So it's okay that I changed them?
- This work has uncovered a subtle bug with pixel => value logic: https://github.com/palantir/blueprint/issues/1769.

#### Screenshot

____SLIDER____

____RANGE SLIDER____

### CODE REVIEW

**Comment by cmslewis** [comment]
FYI this is delayed due to some tricky logic that behaves incorrectly only in the bottom-up coordinate system used in the vertical orientation.

**Comment by cmslewis** [comment]
Removing reviewers for now.

**Comment by cmslewis** [comment]
@tobilen FYI your `DateRangeInput` [tests](https://github.com/palantir/blueprint/pull/1647/files#diff-323ba28e2783ea39a76dda6c3eb4dc10R175
) from #1647 are failing as of this month, because the default `maxDate` is set to `Dec 31` of the current year, and the lack of a `defaultValue` means there's no "next" button to click.

image

Related to #877.

**Comment by cmslewis** [comment]
@llorca done

image

File: packages/core/examples/rangeSliderExample.tsx [hunk 1]
```diff
@@ -28,10 +32,17 @@ export class RangeSliderExample extends BaseExample<IRangeSliderExampleState> {
...
                     value={this.state.range}
+                    vertical={this.state.vertical}
                 />
             </div>
         );
     }
 
+    protected renderOptions() {
+        return [
+            [<Switch checked={this.state.vertical} label={"Vertical"} key="vertical" onChange={this.toggleVertical} />],
```
Comment (giladgray) [inline packages/core/examples/rangeSliderExample.tsx]: `label="Vertical"`

**Pull request merged**